packages/github/src/github.ts

```diff
@@ -17,11 +22,113 @@ export const context = new Context.Context()
 export class GitHub extends Octokit {
   graphql: GraphQL
 
-  constructor(token: string, opts: Omit<Octokit.Options, 'auth'> = {}) {
-    super({...opts, auth: `token ${token}`})
+  /* eslint-disable no-dupe-class-members */
+  // Disable no-dupe-class-members due to false positive for method overload
+  // https://github.com/typescript-eslint/typescript-eslint/issues/291
 
-    this.graphql = graphql.defaults({
-      headers: {authorization: `token ${token}`}
-    })
+  /**
```

Comment:
In addition to setting proxy, I changed the interface so other auth mechanisms can be used. For example see [this issue](https://github.com/actions/toolkit/issues/106).

When customers are using other auth settings, they should still be able to leverage the toolkit to setup proxy.